### PR TITLE
Dataproc config for serverless in profile and fix quoting

### LINKED ISSUE #350: [CT-1336] [Feature] allow setting networkUri and subnetworkUri for Dataproc Serverless batches
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt-bigquery functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

In addition to the existing Dataproc Serverless settings, allow passing through the `network` and `subnetworkUri` fields in the job configuration: https://cloud.google.com/dataproc-serverless/docs/reference/rest/v1/projects.locations.batches#executionconfig

### Describe alternatives you've considered

n/a

### Who will this benefit?

dbt users on BigQuery running Python models with Dataproc Serverless. in particular, enterprise users with enterprise requirements

### Are you interested in contributing this feature?

maybe; also a good first issue for anyone to pick up 

### Anything else?

we can provide guidance if needed

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
resolves #350 

### Description
* Pull in changes to fix quote policy: https://github.com/dbt-labs/dbt-core/issues/7114
* Have functioning integration test 
* Pull in changes from: https://github.com/dbt-labs/dbt-bigquery/pull/381

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-bigquery/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-bigquery/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # install latest changes in dbt-core
 # TODO: how to automate switching from develop to version branches?
-git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-core&subdirectory=core
-git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-tests-adapter&subdirectory=tests/adapter
+git+https://github.com/dbt-labs/dbt-core.git@addPyRelationNameMacro#egg=dbt-core&subdirectory=core
```
Comment (McKnight-42) [inline dev-requirements.txt:3]: nit: dbt-core pointer once approved

File: tests/unit/test_bigquery_adapter.py [hunk 2]
```diff
@@ -184,6 +221,25 @@ def test_acquire_connection_oauth_validations(self, mock_open_connection):
         connection.handle
         mock_open_connection.assert_called_once()
 
+    @patch('dbt.adapters.bigquery.connections.get_bigquery_defaults', return_value=('credentials', 'project_id'))
+    @patch('dbt.adapters.bigquery.BigQueryConnectionManager.open', return_value=_bq_conn())
+    def test_acquire_connection_dataproc_serverless(self, mock_open_connection, mock_get_bigquery_defaults):
```
Comment (McKnight-42) [inline tests/unit/test_bigquery_adapter.py:226]: nit: does this go beyond flake8 and black line length?

**Review by ChenyuLInx — APPROVED** [review]
LGTM, thanks for carry this through!!

**Review by nssalian — APPROVED** [review]

**Comment by colin-rogers-dbt** [comment]
@cla-bot

**Comment by colin-rogers-dbt** [comment]
@cla-bot check

**Pull request merged**